.changeset/early-sloths-deliver.md

```diff
@@ -0,0 +1,46 @@
+---
+'skuba': patch
+---
+
+template: Install specific pnpm version via Corepack
+
+Previously, our Dockerfiles ran `corepack enable pnpm` without installing a specific version. This does not guarantee installation of the pnpm version specified in `package.json`, which could cause a subsequent `pnpm install --offline` to run Corepack online or otherwise hang on stdin:
+
+```dockerfile
+FROM --platform=arm64 node:20-alpine
+
+RUN corepack enable pnpm
+```
+
+```json
+{
+  "packageManager": "pnpm@8.15.4",
+  "engines": {
+    "node": ">=20"
+  }
+}
+```
+
+```console
+Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-8.15.4.tgz.
+
+Do you want to continue? [Y/n]
+```
+
+To avoid this issue, modify (1) Buildkite pipelines to cache on `package.json`, and (2) Dockerfiles to mount `package.json` and run `corepack install`:
+
+```diff
+  seek-oss/docker-ecr-cache#v2.1.0:
+    cache-on:
+     - .npmrc
++    - package.json
+     - pnpm-lock.yaml
+```
```

Comment:
@samchungy I won't release this until we sort out seek-oss/docker-ecr-cache-buildkite-plugin#51, so you can modify this changeset later.

Alternatively, we could actually revert #1512. It _feels_ wrong but `pnpm install --offline` can invoke the Corepack shim and install the appropriate version as required. The reason one of my repos hit a stdin hang is because it wasn't using `propagate-environment` so Corepack thought it wasn't running in `CI=true`.

Reply:
I defer to sam

Reply:
Nah I think what you've got here makes more sense. If we're pinning the version of pnpm, we want to ensure it is using the pinned version, just in case a bad release creeps in.

Reply:
I can probably write up a patch to target this